frontend/components/Main/MainHeader/AgentButton/index.tsx

```diff
@@ -0,0 +1,273 @@
+import { InfoCircleOutlined } from '@ant-design/icons';
+import { Button, ButtonProps, Flex, Popover, Typography } from 'antd';
+import { useCallback, useMemo } from 'react';
+
+import { Chain, DeploymentStatus } from '@/client';
+import { COLOR } from '@/constants/colors';
+import { useBalance } from '@/hooks/useBalance';
+import { useElectronApi } from '@/hooks/useElectronApi';
+import { useServices } from '@/hooks/useServices';
+import { useServiceTemplates } from '@/hooks/useServiceTemplates';
+import { useStakingContractInfo } from '@/hooks/useStakingContractInfo';
+import { useStore } from '@/hooks/useStore';
+import { useWallet } from '@/hooks/useWallet';
+import { ServicesService } from '@/service/Services';
+import { WalletService } from '@/service/Wallet';
+import { getMinimumStakedAmountRequired } from '@/utils/service';
+
+import { CannotStartAgent } from '../CannotStartAgent';
+import { requiredGas, requiredOlas } from '../constants';
+
+const { Text } = Typography;
+
+const LOADING_MESSAGE =
+  "Starting the agent may take a while, so feel free to minimize the app. We'll notify you once it's running. Please, don't quit the app.";
+
+const AgentStartingButton = () => (
+  <Popover
+    trigger={['hover', 'click']}
+    placement="bottomLeft"
+    showArrow={false}
+    content={
+      <Flex vertical={false} gap={8} style={{ maxWidth: 260 }}>
+        <Text>
+          <InfoCircleOutlined style={{ color: COLOR.BLUE }} />
+        </Text>
+        <Text>{LOADING_MESSAGE}</Text>
+      </Flex>
+    }
+  >
+    <Button type="default" size="large" ghost disabled loading>
+      Starting...
+    </Button>
+  </Popover>
+);
+
+const AgentStoppingButton = () => (
+  <Button type="default" size="large" ghost disabled loading>
+    Stopping...
+  </Button>
+);
+
+const AgentRunningButton = () => {
+  const { showNotification } = useElectronApi();
+  const { service, setIsServicePollingPaused, setServiceStatus } =
+    useServices();
+
+  const handlePause = useCallback(async () => {
+    if (!service) return;
+    // Paused to stop overlapping service poll while waiting for response
+    setIsServicePollingPaused(true);
+
+    // Optimistically update service status
+    setServiceStatus(DeploymentStatus.STOPPING);
+    try {
+      await ServicesService.stopDeployment(service.hash);
+    } catch (error) {
+      console.error(error);
+      showNotification?.('Error while stopping service');
```

Comment:
do we need to say "agent" instead of "service"? or they are both synonyms to the user?